alphagov/publishing-api
# Reorganise test code

#### why?
- we need to introduce a draft content store in our specs, so
  - avoid repetition, which will only increase with a draft content store
  - extract testhelpers to make it easier to have more test servers
- `/publish-intent` and `/content` endpoint behaviours are about to diverge

#### what?

- split integration_test.go according to resources
- extracted `integration_test` package to make code dry
  - contains integration tests
  - contains common assertion code
  - keep `TestRequestTracker` code separate
  - use `structs` to establish a contract for what's needed to perform assertions, or setup stub responses 
- retain some test code to make sure test code looks complete, and not too DRY
- make test data meaningful
- using `ghttp.Server`, and its assertions helped realise that we don't set the appropriate content-type on requests sent to URL Arbiter, so added that.
